rust-toolchain.toml

```diff
@@ -0,0 +1,6 @@
+# Specify which version of rust should be used to compile the project
+# cargo + rustup will use this to consistently build the project
+# with the same version across all checkouts and environments
+[toolchain]
```

Comment:
Why is this file needed in addition to the "with: toolchain ..." directive in the cargo.yml file?

Reply:
CI yaml only deterimines which version of rust to install, and on a machine different versions of rust can be installed. 
This file determines for this repo what rust version to use to compile.
If on the machine, a newer version of rust is preinstalled in the image, the ci can break again without this file, since by default the latest version installed is picked.